web/src/main/java/org/springframework/security/web/server/csrf/CsrfWebFilter.java

```diff
@@ -187,7 +187,7 @@ public static void skipExchange(ServerWebExchange exchange) {
 		@Override
 		public Mono<MatchResult> matches(ServerWebExchange exchange) {
 			return Mono.just(exchange.getRequest())
-				.map(r -> r.getMethod())
+				.map(r -> (r.getMethod() != null ? r.getMethod() : Mono.empty()))
```

Comment:
This doesn't look right to me. You shouldn't map to a publisher. It should be a `.flatMap` and both code paths need to be a `Publisher`

```suggestion
				.flatMap(r -> Mono.justOrEmpty(r.getMethod()))
```

Reply:
Just to understand the required change, my understanding was the return from "getMethod" in question is time bounded, not a value for future, therefore it would be okay to use condition in map.

Please enlighten me if I am missing something.

Reply:
The issue is that if `r.getMethod()` is `null` then, it is a `.map(Mono.empty())` means that it is now a `Mono<Mono<Empty>>`. Instead you want just `Mono.empty()`.

Reply:
totally make sense! thanks for clarifying.